src/ripple/app/misc/FeeVoteImpl.cpp

```diff
@@ -217,10 +229,22 @@ FeeVoteImpl::doVoting(
             {
                 obj[sfAccount] = AccountID();
                 obj[sfLedgerSequence] = seq;
-                obj[sfBaseFee] = baseFee;
-                obj[sfReserveBase] = baseReserve;
-                obj[sfReserveIncrement] = incReserve;
-                obj[sfReferenceFeeUnits] = feeUnits;
+                // These throws are likely to terminate the app if any fire.
+                // They should not be possible to hit, though, because the voting
+                // process (FeeVoteImpl) ignores any out of range values.
+                if(auto const v = baseFee.dropsAs<std::uint64_t>())
+                    obj[sfBaseFee] = *v;
+                else
+                    Throw<std::runtime_error>("XRPAmount conversion out of range");
```

Comment:
This is a new exception and I'm not sure that we ever catch this. Can you point to the catch handler @ximinez?

Reply:
There is no catch handler. The original intent was that if the values here are not valid, then something has gone horribly wrong, so the intention was to allow the throw to crash the program.

However, I've given your feedback some thought, and came up with two alternate approaches that I think address all of your concerns
1. https://github.com/ripple/rippled/commit/7ba8fbbf5a9ae22f9ef42ce54fd481f1b84b832a (Comparison to develop: https://github.com/ripple/rippled/compare/develop...ximinez:strong-xrpfee). This change makes a new class, `XRPFee`, which is *derived* from `XRPAmount`, and adds the functionality that isn't appropriate for `XRPAmount` including: division, `dropsAs`. `decimalXRP`, `jsonClipped`, and the `>>` operator.  This allows us to continue to use `XRPAmount` as _the_ class to represent XRP / drops, while still having functionality to represent fees when we need to. It also gets rid of these conversion throws by allowing `dropsAs` to return a default value if the stored value is out of range, and uses the fee values from the LCL as those defaults.
2. https://github.com/ripple/rippled/commit/ce568a5dfc405baeddb552508d49424d0fa25805 (Comparison to develop: https://github.com/ripple/rippled/compare/develop...ximinez:strong-template). This change basically puts `XRPAmount` back in protocol (with a few improvements), and creates `XRPFee` as an instance of `TaggedFee`, and then converts values as necessary. This approach has already been rejected once thanks to @seelabs, but it does also get rid of the throws because the fees are always the right type throughout the voting process. I also don't like this approach, because there are so many conversions between the two types that I worry about one of them being done incorrectly.

So, @nbougalis, @seelabs, thoughts?

Reply:
@ximinez What do you think about removing the division, mod, increment, and decrement functions from XRPAmount? Here's a patch that does so: https://github.com/seelabs/rippled/commit/6cf2c3802ed3bef6f75648750212bf3db2a0812e

The only place that loses some of the type safety you added was the lambda in `scaleFeeLoad`, but I think that's OK, and on balance I think that's worth it to remove those operators (which I agree with @nbougalis, I'd rather they weren't part of the XRPAmount class).

I did a patch that got rid of the multiplication operator, but I think that operator is worth keeping. We only allow multiplication by a scalar, and it's useful for calculating reserves (multiply by owner count), scaling fees, as converting "whole xrps" to drops (tho that could easily be a function if we wanted).

See what you think of this patch, and if @nbougalis is OK with keeping multiplication, I'd rather do something like this than derive a subclass from `XRPAmount`.

Note: this does not solve the issues with throwing, but does solve the operators problem.